acvm/src/pwg/arithmetic.rs

```diff
@@ -127,26 +128,41 @@ impl ArithmeticSolver {
         // We are assuming it has been optimized.
         match arith_gate.mul_terms.len() {
             0 => MulTerm::Solved(FieldElement::zero()),
-            1 => {
-                let q_m = &arith_gate.mul_terms[0].0;
-                let w_l = &arith_gate.mul_terms[0].1;
-                let w_r = &arith_gate.mul_terms[0].2;
-
-                // Check if these values are in the witness assignments
-                let w_l_value = witness_assignments.get(w_l);
-                let w_r_value = witness_assignments.get(w_r);
-
-                match (w_l_value, w_r_value) {
-                    (None, None) => MulTerm::TooManyUnknowns,
-                    (Some(w_l), Some(w_r)) => MulTerm::Solved(*q_m * *w_l * *w_r),
-                    (None, Some(w_r)) => MulTerm::OneUnknown(*q_m * *w_r, *w_l),
-                    (Some(w_l), None) => MulTerm::OneUnknown(*q_m * *w_l, *w_r),
-                }
-            }
+            1 => ArithmeticSolver::solve_mul_term_helper(
+                &arith_gate.mul_terms[0],
+                witness_assignments,
+            ),
             _ => panic!("Mul term in the arithmetic gate must contain either zero or one term"),
```

Comment:
We shoul create an issue to see if it is correct to panic here for R1CS, ie assuming that we can only do a multiplication with 2 terms or less

Reply:
This is incorrect for R1CS, only the CSatOptimizer is keeping one multiplication.

Reply:
Noting for reference: The `solve_mul_term_helper` and `evaluate` methods are added because of this 2 terms issue.

Have asked Patrick whether this is okay. This is a separate issue though, so its not blocking